asv_benchmarks/asv.conf.json

```diff
@@ -0,0 +1,163 @@
+    // the empty string, the project is assumed to be located at the root
+    // of the repository.
+    // "repo_subdir": "",
+
+    // Customizable commands for building, installing, and
+    // uninstalling the project. See asv.conf.json documentation.
+    "install_command": ["python -mpip install {wheel_file}"],
+    "uninstall_command": ["return-code=any python -mpip uninstall -y {project}"],
+    "build_command": ["python -m build --wheel -o {build_cache_dir} {build_dir}"],
+
+    // List of branches to benchmark. If not provided, defaults to "master
+    // (for git) or "default" (for mercurial).
+    "branches": ["main"],
+    // "branches": ["default"],    // for mercurial
+
+    // The DVCS being used.  If not set, it will be automatically
+    // determined from "repo" by looking at the protocol in the URL
+    // (if remote), or by looking for special directories, such as
+    // ".git" (if local).
+    // "dvcs": "git",
+
+    // The tool to use to create environments.  May be "conda",
+    // "virtualenv" or other value depending on the plugins in use.
+    // If missing or the empty string, the tool will be automatically
+    // determined by looking for tools on the PATH environment
+    // variable.
+    "environment_type": "conda",
+
+    // timeout in seconds for installing any dependencies in environment
+    // defaults to 10 min
+    //"install_timeout": 600,
+
+    // the base URL to show a commit for the project.
+    "show_commit_url": "https://github.com/nilearn/nilearn/commit/",
+
+    // The Pythons you'd like to test against. If not provided, defaults
+    // to the current version of Python used to run `asv`.
+    "pythons": ["3.9"],
+
+    // The list of conda channel names to be searched for benchmark
+    // dependency packages in the specified order
+    // "conda_channels": ["conda-forge", "defaults"]
+
+    // The matrix of dependencies to test. Each key is the name of a
+    // package (in PyPI) and the values are version numbers. An empty
+    // list or empty string indicates to just test against the default
+    // (latest) version. null indicates that the package is to not be
+    // installed. If the package to be tested is only available from
+    // PyPi, and the 'environment_type' is conda, then you can preface
+    // the package name by 'pip+', and the package will be installed via
+    // pip (with all the conda available packages installed first,
+    // followed by the pip installed packages).
+    //
+    // The versions of the dependencies should be bumped in a dedicated commit
+    // to easily identify regressions/improvements due to code changes from
+    // those due to dependency changes.
+    //
+    "matrix": {
+        "numpy": ["1.22.4"],
+        "scipy": ["1.8.0"],
+        "joblib": ["1.2.0"],
+        "pandas": ["2.2.0"],
+        "scikit-learn": ["1.4.0"],
+        "nibabel": ["5.2.0"],
+    }
+
+    // Combinations of libraries/python versions can be excluded/included
```

Comment:
Are these lines to be removed ?

Reply:
Those come from the default config generated by asv. I think we should keep them if we want to use some that functionality later on.

Reply:
Yep, for now, this config is only being read by asv and none of our own scripts, so we don't need any additional dependencies.